File: blackjack_draw.py
import random

# simulate the initial deck and card values
card_types = ("ace", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "jack", "queen", "king")
deck = {}
for i in card_types:
    deck[i] = 4

card_values = {}
for i in range(1, 11):
    card_values[card_types[i]] = i
card_values["jack"] = 10
card_values["queen"] = 10
card_values["king"] = 10


# deal a hand of 2 cards
hand = ()
for i in range(2):
    card = (card_types[random.randint(0, 12)], )
    deck[card[0]] -= 1
    hand += card

print("current hand:", hand)

# hit to recieve a new card
hit = (card_types[random.randint(0, 12)], )
deck[hit[0]] -= 1

print("hit card:", hit)

hand += hit
print("new hand:", hand)

File: blackjack_draw.r
# simulate the initial deck of 52
card_values <- c("ace", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "jack", "queen", "king")
deck <- rep(card_values, 4)

# initialize the hand values we will test
tested_hand_values <- c(11:20)

# start with a hand of 2 cards from the deck
hand <- sample(card_values, size = 2, replace = FALSE)
cat("starting hand: ", hand, "\n")

# map the deck cards to their numerical values


# "hit" and gain one more card
hit_card <- sample(card_values, size = 1, replace = FALSE)
cat("hit card: ", hit_card, "\n")

hand <- append(hand, hit_card)
cat("new hand: ", hand, "\n")